substratevm/mx.substratevm/suite.py

```diff
@@ -326,6 +326,43 @@
             "workingSets": "SVM",
         },
 
+        "com.oracle.svm.core.jfr": {
```

Comment:
Should there be an entry for `com.oracle.svm.hosted.jfr as well? (maybe similar to the entry for `com.oracle.svm.hosted.jdk11`)

Reply:
I am not sure here.  Adding com.oracle.svm.core.jfr entry here, because it has extra dependencies, etc.

I believe that com.oracle.svm.hosted.jfr is different from com.oracle.svm.hosted.jdk11. com.oracle.svm.hosted.jdk11 has its own directory. Also, I don't see entries for sub-packages of com.oracle.svm.hosted, e.g. com.oracle.svm.hosted.jdk, etc.
 
Honestly, I don't quite understand how suite.py works. I would appreciate if you can point me any doc/wiki, etc.

Reply:
The documentation is located at https://github.com/graalvm/mx

You need to do the following steps:
- Add a project `com.oracle.svm.core.jfr` (analogue to `com.oracle.svm.jfr`) but with only one dependency on `com.oracle.svm.core`
- Add a project `com.oracle.svm.hosted.jfr` (analogue to `com.oracle.svm.jfr`) but with only one dependency on `com.oracle.svm.hosted`
- Add `com.oracle.svm.core.jfr` as a dependency to `com.oracle.svm.core.genscavenge`
- Add the projects that you added to the distribution `SVM`
- Remove the project `com.oracle.svm.jfr`
- Do `mx clean` and `mx ideclean`
- Test everything with `mx build` and `mx ideinit`. It should still be possible to build everything with `mx build` and the IDE should be happy as well (once you import the new project files generated by `mx ideinit`).

Reply:
@christianhaeubl Thanks for the instructions. 

Now, I am running into Java compliance issue, as com.oracle.svm.core.genscavenge has dependence on com.oracle.svm.core.jfr with higher Java compliance.

Should I move com.oracle.svm.core.jfr  under com.oracle.svm.core.jdk11 and create interface, e.g. JfrSupport to workaround jdk8?

Thanks.

Reply:
I see, then the only option is to create a proper dummy implementation for JDK8. We would need at least 4 JFR projects then:
- `com.oracle.svm.core.jfr` (common infrastructure and interfaces)
- `com.oracle.svm.core.jfr.jdk8` (a dummy implementation that always acts as if JFR support is disabled)
- `com.oracle.svm.core.jfr.jdk11` (the current implementation from `com.oracle.svm.jfr`)
- `com.oracle.svm.hosted.jfr` (assuming that there is nothing JDK-specific in those classes)

`com.oracle.svm.core.genscavenge` would then only have a dependency on the JDK-independent part, i.e., `com.oracle.svm.core.jfr`.